ts/input/tex/unicode/UnicodeConfiguration.ts

```diff
@@ -89,8 +90,83 @@ UnicodeMethods.Unicode = function(parser: TexParser, name: string) {
   parser.Push(node);
 };
 
+/**
+ * Create a raw unicode character in TeX input.
+ *
+ * @param {TexParser} parser The current tex parser.
+ * @param {string} name The name of the macro.
+ */
+UnicodeMethods.RawUnicode = function (parser: TexParser, name: string) {
+  const hex = parser.GetArgument(name).trim();
+  if (!hex.match(/^[0-9A-F]{1,6}$/)) {
+    throw new TexError('BadRawUnicode',
+                       'Argument to %1 must a hexadecimal number with 1 to 6 digits', parser.currentCS);
+  }
+  const n = parseInt(hex, 16);
+  parser.string = String.fromCodePoint(n) + parser.string.substr(parser.i);
+  parser.i = 0;
+}
+
+/**
+ * Implements the \char control sequence.
+ *
+ * @param {TexParser} parser The current tex parser.
+ * @param {string} name The name of the macro.
+ */
+UnicodeMethods.Char = function (parser: TexParser, _name: string) {
+  let match;
+  let next = parser.GetNext();
+  let c = '';
+  const text = parser.string.substr(parser.i);
+  if (next === '\'') {
+    match = text.match(/^'(?:([0-7]{1,7}) ?|(\\\S)|(.))/u);
+    if (match) {
```

Comment:
Can we `break` here on `!match` to reduce nesting depth.

Reply:
No, I don't think so.  There is no loop or `switch` statement to terminate, and we do want to fall through to the `if (!c)` that produces the error message at line 150.  So I don't think `break` works here.